1-Vanilla/scripts/todo_list.js

```diff
@@ -45,32 +51,36 @@
   }
 
   function addItem() {
-    let newItemInput = document.getElementById("new-item-name");
-
     if (isVaildItemName(newItemInput.value)) {
-      let newItem = createTodoItem(nextId, newItemInput.value);
+      const newItem = createTodoItem(nextId, newItemInput.value);
 
       nextId++;
 
       document.getElementById("main-container").appendChild(newItem);
-      document.getElementById("new-item-name").value = "";
+      newItemInput.value = "";
     }
   }
 
   function isVaildItemName(name) {
-      return name.length > 0;
+      let valid = name.length > 0;
+
+      if (!valid) {
```

Comment:
`alert` doesn't provide a very good UX 😄  i'd go with a better UX solution such as disabling the button, suppressing the action or so